src/Task/ConvertTranslatableTask.php

```diff
@@ -0,0 +1,268 @@
+        if (empty($locales)) {
+            throw new Exception("Please configure Fluent locales (in the CMS) prior to migrating from translatable");
+        }
+
+        $defaultLocale = Locale::getDefault();
+        if (empty($defaultLocale)) {
+            throw new Exception(
+                "Please configure a Fluent default locale (in the CMS) prior to migrating from translatable"
+            );
+        }
+    }
+
+    /**
+     * Do something inside a DB transaction
+     *
+     * @param callable $callback
+     * @throws \Exception
+     */
+    protected function withTransaction($callback)
+    {
+        try {
+            Debug::message('Beginning transaction', false);
+            DB::get_conn()->transactionStart();
+            $callback($this);
+            Debug::message('Comitting transaction', false);
+            DB::get_conn()->transactionEnd();
+        } catch (\Exception $ex) {
+            Debug::message('Rolling back transaction', false);
+            DB::get_conn()->transactionRollback();
+            throw $ex;
+        }
+    }
+
+    protected $translatedFields = [];
+
+    /**
+     * Get all database fields to translate
+     *
+     * @param string $class Class name
+     * @return array List of translated fields
+     */
+    public function getTranslatedFields($class)
+    {
+        if (isset($this->localisedFields[$class])) {
+            return $this->localisedFields[$class];
+        }
+        $fields = [];
+        $hierarchy = ClassInfo::ancestry($class);
+        foreach ($hierarchy as $class) {
+            // Skip classes without tables
+            if (!DataObject::getSchema()->classHasTable($class)) {
+                continue;
+            }
+
+            // Check fields localised by Fluent for this class
+            $localisedFields = singleton($class)->getLocalisedFields();
+            if (empty($localisedFields)) {
+                continue;
+            }
+
+            // Save fields
+            $fields = array_merge($fields, array_keys($localisedFields));
+        }
+        $this->localisedFields[$class] = $fields;
+        return $fields;
+    }
+
+    /**
+     * Gets all classes with FluentExtension
+     *
+     * @return array Array of classes to migrate
+     */
+    public function fluentClasses()
+    {
+        $classes = [];
+        $dataClasses = ClassInfo::subclassesFor(DataObject::class);
+        array_shift($dataClasses);
+        foreach ($dataClasses as $class) {
+            $base = DataObject::getSchema()->baseDataClass($class);
+            foreach (DataObject::get_extensions($base) as $extension) {
+                if (is_a($extension, FluentExtension::class, true)) {
+                    $classes[] = $base;
+                    break;
+                }
+            }
+        }
+        return array_unique($classes);
+    }
+
+    public function run($request)
+    {
+        // Extend time limit
+        set_time_limit(100000);
```

Comment:
This isn't a wise idea IMO - PHP's CLI SAPI has a timelimit of `0` (inifinite) _by default_ - unless this is set at run time during SS's boot (which would be a bit foolish IMO) we shouldn't 'increase' the limit from `infinity` to `some_large_but_still_finite_number`.

I've been burnt by jobs doing this in the past, and it is certainly both tiresome to debug (looking for errors as opposed to timeouts), and 4x as unpleasant to clean up after (albeit you're using transactions here so hopefully it wouldn't' be so bad). When execution stops, it simply stops - with no cleanup IIRC.

Reply:
Fair comment. This is here from the original SS3 task's code

Reply:
Do you have a suggestion for how to implement this in a way that will work for the CLI and via web as well? I realise servers like Apache may have their own timeouts, but the process will still continue to run in the background. I don't like the idea of creating a script that can run infinitely in the background (if initiated from the web)

Reply:
By default; simply do not attempt to address this (non) issue.

Each PHP SAPI has it's own configuration, and by default this is set to a positive integer (which I cannot recall) for Apache (and other non-CLI SAPIs afaik). So unless a system administrator has made a less than advisable adjustment to the php.ini for that sapi, this should be fine.

Reply:
Looking back through the git history there doesn't seem to be an explicit reason for this to be here, so unless @tractorcow disagrees I'm removing it now